### PR TITLE
resource/aws_kinesis_firehose_delivery_stream: Support Redshift processing_configuration

### LINKED ISSUE #513: Support Kinesis Firehose Data Transformation
_This issue was originally opened by @mlimaloureiro as hashicorp/terraform#11723. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi Guys,

I would like to request Terraform to support `Kinesis Firehose` data transformation.

More info here

https://aws.amazon.com/about-aws/whats-new/2016/12/amazon-kinesis-firehose-can-now-prepare-and-transform-streaming-data-before-loading-it-to-data-stores/ 

Thanks!

### PR DESCRIPTION
**Pull request by bflad**
Closes #513 

Test failure is a known issue.
```
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3InvalidParameterName
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3InvalidParameterName (1.53s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3InvalidProcessorType
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3InvalidProcessorType (1.95s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_missingProcessingConfiguration
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_missingProcessingConfiguration (97.07s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_s3KinesisStreamSource (102.50s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3Updates
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3Updates (103.84s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3basic
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3basic (107.25s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_importBasic
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_importBasic (108.98s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_s3basic
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_s3basic (109.01s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3KmsKeyArn
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_ExtendedS3KmsKeyArn (125.83s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_s3ConfigUpdates
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_s3ConfigUpdates (183.52s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_s3WithCloudwatchLogging
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_s3WithCloudwatchLogging (229.20s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_SplunkConfigUpdates
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_SplunkConfigUpdates (262.68s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_RedshiftConfigUpdates
--- PASS: TestAccAWSKinesisFirehoseDeliveryStream_RedshiftConfigUpdates (1058.01s)
=== RUN   TestAccAWSKinesisFirehoseDeliveryStream_ElasticsearchConfigUpdates
--- FAIL: TestAccAWSKinesisFirehoseDeliveryStream_ElasticsearchConfigUpdates (1273.59s)
    testing.go:518: Step 0 error: Error applying: 1 error(s) occurred:
        
        * aws_kinesis_firehose_delivery_stream.test_stream_es: 1 error(s) occurred:
        
        * aws_kinesis_firehose_delivery_stream.test_stream_es: error creating Kinesis Firehose Delivery Stream: InvalidArgumentException: Delivery Role is not authorized to describe the Elasticsearch domain: arn:aws:es:us-west-2:*******:domain/es-test-8013883722251758873.
            status code: 400, request id: 4c708fc7-432b-11e8-9cb3-a9462fe68d6e
```

### CODE REVIEW

**Comment by scorsi** [comment]
Hi,
In which version will this feature be implemented ? Or maybe is it currently available ?

**Comment by bflad** [comment]
@scorsi Generally pull requests will be marked with the appropriate vX.Y.Z milestone if they are targeting a specific release. This pull request will have that information when it is reviewed by a maintainer and merged, but it is currently not targeting any specific release.

File: aws/resource_aws_kinesis_firehose_delivery_stream_test.go [hunk 1]
```diff
@@ -1177,6 +1202,16 @@ resource "aws_kinesis_firehose_delivery_stream" "test_stream" {
     data_table_name = "test-table"
     copy_options = "GZIP"
     data_table_columns = "test-col"
+    processing_configuration = [{
+      enabled = "false",
```
Comment (tombuildsstuff) [inline aws/resource_aws_kinesis_firehose_delivery_stream_test.go]: does this need the quotes around it?

Comment (bflad) [reply aws/resource_aws_kinesis_firehose_delivery_stream_test.go]: It does not! Thanks for catching that.

**Comment by bflad** [comment]
This has been released in [version 1.16.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1160-april-25-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**